### DISCUSSION TITLE
@JoinColumn 과 mappedBy 속성에 대해 알려주세요.

### DISCUSSION THREAD

**Discussion by gyungchan-jo** [2024-woowacourse-study/level-interview#224] [Spring Data]
.

**Comment by le2sky** [comment]
`@JoinColumn`은 외래키를 매핑할 때 사용합니다.

다음과 같은 속성을 가지고 있습니다.

- name : 매핑할 외래 키 이름(필드명 + _ + 참조하는 테이블의 기본 키 컬럼명)
- referencedCloumnName : 외래 키가 참조하는 대상 테이블의 컬럼명 (기본값 : 참조하는 테이블의 기본 키 컬럼명)
- foreignKey : 외래 키 제약조건을 직접 지정할 수 있습니다. 테이블 생성시에만 사용됩니다.
- unique, nullable, insertable, updatable, columnDefinition, table..

mappedBy 속성은 연관관계의 주인을 정할 경우 사용합니다. 주인이 아닌 쪽에는 mappedBy 속성을 사용해서 속성의 값으로 연관관계의 주인을 지정해야합니다.

**Comment by J-I-H-O** [comment]
`@JoinColumn` 은 외래 키를 매핑할 때 사용합니다. `name` 속성에는 매핑할 외래 키 컬럼명을 지정합니다. 
`@JoinColumn` 은 생략할 수 있는데, 생략 시 외래키를 찾을 때 `필드명_[참조하는 테이블의 기본 키 컬럼명]` 의 기본 전략을 사용합니다.

`mappedBy`  연관관계의 주인을 지정해주는 속성입니다. 
주인이 아닌 쪽에 `mappedBy` 속성을 사용하며, 연관관계 주인의 해당 속성의 필드명을 지정해줍니다.

https://ttl-blog.tistory.com/126
